packages/apps/hr/src/components/molecules/ExpenseAddPopup/ExpenseAddPopup.js

```diff
@@ -26,36 +26,82 @@ import {
   Picker,
 } from '@axelor/aos-mobile-ui';
 import {useTranslator, useSelector, useDispatch} from '@axelor/aos-mobile-core';
-import {searchExpenseDraft} from '../../../features/expenseSlice';
+import {
+  createExpense,
+  fetchExpenseById,
+  searchExpenseDraft,
+  updateExpense,
+} from '../../../features/expenseSlice';
 
-const ExpenseAddPopup = ({style, visible, onClose}) => {
+const ExpenseAddPopup = ({
+  style,
+  visible,
+  onClose,
+  selectedItems,
+  navigation,
+}) => {
   const I18n = useTranslator();
   const Colors = useThemeColor();
   const dispatch = useDispatch();
 
   const {expenseDraftList} = useSelector(state => state.expense);
+  const {expense} = useSelector(state => state.expense);
+  const {user} = useSelector(state => state.user);
 
   const [expenseSelected, setExpenseSelected] = useState(null);
 
   useEffect(() => {
     dispatch(searchExpenseDraft());
   }, [dispatch]);
 
+  useEffect(() => {
+    if (expenseSelected != null) {
+      dispatch(fetchExpenseById({ExpenseId: expenseSelected}));
+    }
+  }, [dispatch, expenseSelected]);
+
+  const createExpenseAPI = useCallback(() => {
+    const _expense = {
+      expenseLineIdList: selectedItems,
+      employeeId: user?.employee?.id,
+      companyId: user?.activeCompany?.id,
+      currencyId: user?.activeCompany?.currency?.id,
+    };
+    dispatch(createExpense({expense: _expense, userId: user.id}));
+    onClose();
+    navigation.navigate('ExpenseListScreen');
+  }, [dispatch, onClose, navigation, selectedItems, user]);
+
+  const updateExpenseAPI = useCallback(() => {
+    dispatch(
+      updateExpense({
+        expenseId: expense.id,
+        version: expense.version,
+        userId: user.id,
+        expenseLineIdList: selectedItems,
+      }),
+    );
+    navigation.navigate('ExpenseListScreen');
+    onClose();
+  }, [dispatch, onClose, expense, navigation, selectedItems, user.id]);
+
   return (
     <PopUp style={[styles.popup, style]} visible={visible}>
       <View style={styles.container}>
         <View style={styles.pickerContainer}>
           <Picker
             pickerStyle={styles.picker}
             listItems={expenseDraftList}
-            onValueChange={setExpenseSelected}
+            onValueChange={e => {
+              setExpenseSelected(e);
+            }}
```

Comment:
revert change, useful to debug but it's dirty
use isValueItem on Picket to get the whole item on selection. You won't need to realize a fetch for the expense this way